lib/application/theme/theme_controller.dart

```diff
@@ -0,0 +1,108 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:memo/application/constants/dimensions.dart' as dimens;
+import 'package:memo/application/theme/material_theme_data.dart' as material_theme;
+import 'package:memo/application/theme/memo_theme_colors.dart' as colors;
+import 'package:memo/application/theme/memo_theme_data.dart';
+
+final themeProvider = StateNotifierProvider<ThemeController>((ref) => ThemeController());
```

Comment:
`ref` is not used in its scope

Reply:
Also, isn't lint complaining about this?

Reply:
I don't think dart has a check for unused parameters inside a particular closure